### ISSUE TITLE
[prefer-readonly-type] Should consider parameters to be non-local

### ISSUE THREAD

**Issue by whatisaphone** [jonaskello/eslint-plugin-functional#75]
Thanks for this plugin and for migrating it to eslint!

I migrated our codebase yesterday and I noticed a possible regression from the tslint plugin. Consider this config and code:

TSLint config: `"readonly-keyword": [true, "ignore-local"]`

ESLint config: `"functional/prefer-readonly-type": ["error", { "allowLocalMutation": true }]`

```ts
function distance(point: {
  x: number;
  y: number;
}) {
  return (point.x * point.x + point.y * point.y) ** 0.5;
}
```

The **TSLint** plugin warns about missing `readonly`, because it considers the type of `point` to be **non-local**. The **ESLint** plugin does not warn, because it considers it to be **local**. I think the TSLint plugin was doing it right. Changes to `point` can be observed outside the function, thus the type is **non-local**, and it should be enforced as `readonly`.

<details>
<summary>Don't try this at home</summary>

```ts
function distance(point: {
  x: number;
  y: number;
}) {
  point.x *= point.x;
  point.y *= point.y;
  return (point.x * point.y) ** 0.5;
}
```

</details>
